### PR TITLE
Add a specification for optional jobs in ci-operator configuration

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/test-platform 

https://issues.redhat.com/browse/DPTP-2501
Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
/hold 

This should break the current config, since there are jobs that have `optional: true` already added by the users manually. Holding this to investigate the safest way to proceed with those changes.

**Comment by droslean** [comment]
/hold cancel

Since we already allow users to edit their `optional` field without breaking the job generation, this should be safe to merge.

File: pkg/api/types.go [hunk 1]
```diff
@@ -650,6 +650,9 @@ type TestStepConfiguration struct {
 	// RunIfChanged is a regex that will result in the test only running if something that matches it was changed.
 	RunIfChanged string `json:"run_if_changed,omitempty"`
 
+	// Optional defines if the job will be marked as optional or required.
```
Comment (petr-muller) [inline pkg/api/types.go:653]: Such Godoc really has no added value over the name of the field, please make it more precise (good sentences will probably mention something about merging)

**Review by petr-muller — COMMENTED** [review]
This is missing some important bits:

1) Please add validation that this is only used on presubmits
2) Please add a treatment like https://github.com/openshift/ci-tools/blob/752705aa4b09d516b21ae614bf28890a552a809d/pkg/jobconfig/files.go#L476-L483 to ensure that if a ci-op config sets the field, it cannot be overriden on the job level.

File: pkg/validation/test.go [hunk 2]
```diff
@@ -161,8 +161,8 @@ func (v *Validator) validateTestStepConfiguration(
 		if test.Interval != nil && test.ReleaseController {
 			validationErrors = append(validationErrors, fmt.Errorf("%s: `interval` cannot be set for release controller jobs", fieldRootN))
 		}
-		if (test.Cron != nil || test.Interval != nil) && (test.RunIfChanged != "" || test.SkipIfOnlyChanged != "") {
-			validationErrors = append(validationErrors, fmt.Errorf("%s: `cron` and `interval` are mutually exclusive with `run_if_changed`/`skip_if_only_changed`", fieldRootN))
+		if (test.Cron != nil || test.Interval != nil) && (test.RunIfChanged != "" || test.SkipIfOnlyChanged != "" || test.Optional) {
+			validationErrors = append(validationErrors, fmt.Errorf("%s: `cron` and `interval` are mutually exclusive with `run_if_changed`/`skip_if_only_changed`/`optional`", fieldRootN))
```
Comment (petr-muller) [inline pkg/validation/test.go:165]: optional is also exclusive with postsubmit

Comment (droslean) [reply pkg/validation/test.go:168]: addressed. Test case also added.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**